tests/Application.Tests.Integration/Folders/Commands/DisableFolderTests.cs

```diff
@@ -0,0 +1,191 @@
+            NumberOfLockers = 1,
+        };
+
+        var locker = new Locker()
+        {
+            Id = Guid.NewGuid(),
+            Capacity = 47,
+            Name = "fqwlkjdb sajdbqwk",
+            IsAvailable = true,
+            Room = room,
+            NumberOfFolders = 1,
+        };
+        
+        var folder = new Folder()
+        {
+            Id = Guid.NewGuid(),
+            Capacity = 12,
+            IsAvailable = true,
+            Name = "A Random name",
+            NumberOfDocuments = 0,
+            Locker = locker
+        };
+
+        await AddAsync(folder);
+        var disableFolderCommand = new DisableFolderCommand()
+        {
+            FolderId = folder.Id
+        };
+        
+        // Act
+        var disabledFolder = await SendAsync(disableFolderCommand);
+
+        // Assert
+        disabledFolder.IsAvailable.Should().BeFalse();
+        
+        // Clean up
+        Remove(folder);
+        Remove(locker);
+        Remove(room);
+    }
+
+    [Fact]
+    public async Task ShouldThrowKeyNotFoundException_WhenFolderDoesNotExist()
+    {
+        // Arrange
+        var disableFolderCommand = new DisableFolderCommand()
+        {
+            FolderId = Guid.NewGuid()
+        };
+        // Act
+        var result = async () => await SendAsync(disableFolderCommand);
+        // Assert
+        await result.Should().ThrowAsync<KeyNotFoundException>()
+            .WithMessage("Folder does not exist.");
+    }
+
+    [Fact]
+    public async Task ShouldThrowKeyInvalidOperationException_WhenFolderAlreadyDisabled()
+    {
+        // Arrange
+        var room = new Room()
+        {
+            Id = Guid.NewGuid(),
+            Capacity = 24,
+            IsAvailable = true,
+            Name = "Kamito's room!",
+            NumberOfLockers = 1,
+        };
+
+        var locker = new Locker()
+        {
+            Id = Guid.NewGuid(),
+            Capacity = 47,
+            Name = "fqwlkjdb sajdbqwk!",
+            IsAvailable = true,
+            Room = room,
+            NumberOfFolders = 1,
+        };
+        
+        var folder = new Folder()
+        {
+            Id = Guid.NewGuid(),
+            Capacity = 12,
+            IsAvailable = false,
+            Name = "A Random name!",
+            NumberOfDocuments = 0,
+            Locker = locker
+        };
+        await AddAsync(folder);
+        var disableFolderCommand = new DisableFolderCommand()
+        {
+            FolderId = folder.Id
+        };
+        
+        // Act
+        var result = async () => await SendAsync(disableFolderCommand);
+
+        // Assert
+        await result.Should().ThrowAsync<InvalidOperationException>()
+            .WithMessage("Folder has already been disabled.");
+        
+        // Clean up
+        Remove(folder);
+        Remove(locker);
+        Remove(room);
+    }
+    
+    [Fact]
+    public async Task ShouldThrowKeyInvalidOperationException_WhenFolderHaveDocuments()
```

Comment:
KeyInvalidOperationException?

Reply:
fixed at https://github.com/ProFile-org/ProFile-BE/pull/88/commits/079c2a2468676481a1e4f5b057a50c9268be09d6